dist/backdraft.js

```diff
@@ -1178,31 +1190,63 @@ _.extend(Plugin.factory, {
       var table = this;
       var filter = col.filter;
 
-      // bind focus to click event because of unbinding click from thead th when
-      // installing sort interceptors
-      $('input', head).on("click", function () {
+      // need to manually set focus because of how we unbound filtering events from the head
+      $('input', head).on('click', function() {
         this.focus();
       });
-      // update columnManager filter and ajaxUpdate dataTable when input changed
+
+      // update columnManager filter when input changed
       $('input', head).on('change', function () {
+        var filterIcon = $('.DataTables_filter_wrapper span', head);
         if (filter.type === "list") {
           if (this.checked) {
             filter[this.id] = filter[this.id] || [];
             filter[this.id].push(this.value);
+            filterIcon.removeClass("filterInactive");
+            filterIcon.addClass("filterActive");
           }
-          else {
+          else if (filter[this.id]) {
             var index = filter[this.id].indexOf(this.value);
             if (index > -1)
               filter[this.id].splice(index, 1);
-            if (filter[this.id].length === 0)
+            if (filter[this.id].length === 0) {
               filter[this.id] = null;
+              filterIcon.removeClass("filterActive");
+              filterIcon.addClass("filterInactive");
+            }
           }
         } else if (this.value === "") {
           filter[this.id] = null;
+          filterIcon.removeClass("filterActive");
+          filterIcon.addClass("filterInactive");
         } else {
           filter[this.id] = this.value;
+          filterIcon.removeClass("filterInactive");
+          filterIcon.addClass("filterActive");
         }
+      });
+
+      // update table when filter button clicked
+      $('.btn-filter', head).on('click', function() {
+        $("input", head).each(function() {
+          if (this.type === "text") {
```

Comment:
maybe I'm missing something but can this also be:

```
$("input[type=text]", head).each(function() {
  $(this).trigger("change");
}
```

and then it simply becomes:

```
$("input[type=text]", head).trigger("change");
```